Sources/Publish/API/DeploymentMethod.swift

```diff
@@ -89,7 +89,75 @@ public extension DeploymentMethod {
     /// - parameter repository: The full name of the repository (including its username).
     /// - parameter useSSH: Whether an SSH connection should be used (preferred).
     static func gitHub(_ repository: String, branch: String = "master", useSSH: Bool = true) -> Self {
+        git(gitHubRemote(repository: repository, useSSH: useSSH),
+            branch: branch)
+    }
+    
+    static func gitHubPages(_ repository: String,
+                            on branch: GitHubPagesDeploymentMode = .master,
+                            useSSH: Bool = true)
+        -> Self
+    {
+        let remote = gitHubRemote(repository: repository, useSSH: useSSH)
+        
+        return DeploymentMethod(name: "GitHub Pages (\(remote)") { context in
+            let jekyllDisablingFile = try context.createOutputFile(at: Path(".nojekyll"))
+            
+            let branchName : String
+            var docsModeFolders : (Folder, Folder)?
+            
+            switch branch {
+            case .ghPages :
+                branchName = "gh-master"
+            case .masterDocs :
+                let docs = try context.createOutputFolder(at: Path("docs"))
+                
+                guard let docsParent = docs.parent else {
+                    try jekyllDisablingFile.delete()
+                    try docs.delete()
+                    return
+                }
+                
+                try docsParent.moveContents(to: docs)
+                
+                docsModeFolders = (docs, docsParent)
+                fallthrough
+            case .master :
+                branchName = "master"
+            }
+            
+            try gitHub(repository,
+                       branch: branchName,
+                       useSSH: useSSH)
+                .body(context)
+            
+            if let (docs, docsParent) = docsModeFolders {
+                try docs.moveContents(to: docsParent)
+                try docs.delete()
+            }
+            
+            try jekyllDisablingFile.delete()
+            
+            let ghPagesModeName : String
+            switch branch {
+            case .master : ghPagesModeName = "master branch"
+            case .masterDocs : ghPagesModeName = "master branch /docs folder"
+            case .ghPages : ghPagesModeName = "gh-pages branch"
+            }
+            
+            let settingsURL = "\(gitHubRemote(repository: repository, useSSH: false))/settings"
```

Comment:
The resulting URL is incorrect. It is:

 https://github.com/a/b.github.io.git/settings

should be:

 https://github.com/a/b.github.io/settings

(In other words you need to trim the ".git" from the url before adding the /settings.

Reply:
By the way, I ran into problems trying to use this patch.  The second time I tried to deploy to the same github project, the deploy step would produce an error because it was trying to add the .nojeckle file a second time.

Also I'm not sure that this extension is even necessary. I was able to use the regular github publisher to publish to my github.io page without any problem.

Reply:
> (In other words you need to trim the ".git" from the url before adding the /settings.

Ok, changed the link generation so it doesn't insert the ".git" part every time. 

>  The second time I tried to deploy to the same github project, the deploy step would produce an error because it was trying to add the .nojeckle file a second time.

Right now checking on this one

> Also I'm not sure that this extension is even necessary

Not all repos use `master` for Github Pages, specially the ones generating a documentation site or a blog related to an already existing repo